Library/Homebrew/linkage_checker.rb

```diff
@@ -21,7 +21,12 @@ def initialize(keg, formula = nil)
     check_dylibs
   end
 
+  def dylib_to_dep(dylib)
+    dylib.sub("#{HOMEBREW_PREFIX}/opt/", "").sub("#{HOMEBREW_CELLAR}/", "").split("/")[0]
```

Comment:
`dylib =~ %r{#{Regexp.escape(HOMEBREW_PREFIX)}/(opt|Cellar)/([\w+-.@]+)/}` and then you can use the second capture group.

Reply:
@maxim-belkin I think it would be good to refactor that and handle the `nil` case instead.